components/selection/selection-checkbox.js

```diff
@@ -62,18 +67,19 @@ class Checkbox extends SkeletonMixin(LitElement) {
 
 	connectedCallback() {
 		super.connectedCallback();
-		this.dispatchEvent(new CustomEvent('d2l-selection-checkbox-subscribe', {
+		const evt = new CustomEvent('d2l-selection-checkbox-subscribe', {
 			bubbles: true,
-			composed: true
-		}));
+			composed: true,
+			detail: {}
+		});
+		this.dispatchEvent(evt);
```

Comment:
The parent selection collection (the list or table that has the mixin) will populate `detail.provider`, which we store so that we can later tell it to unsubscribe us when we're disconnected.

Reply:
We could actually use a different controller for the selection checkboxes and select-all checkbox... maybe a "selection modifier controller"? It could have methods like `selectAll()`, `unselectAll()` and `select(key), `unselect(key)`. Under the covers it would fire these events, but that would make it a bit cleaner to add more UI bits that modify selection down the road.